include/irr/asset/IAsset.h

```diff
@@ -11,63 +11,133 @@ namespace asset
+		ET_SHADER represents
+		ET_SPECIALIZED_SHADER represents
+		ET_MESH_DATA_DESCRIPTOR represents
+		ET_GRAPHICS_PIPELINE represents
+		ET_SCENE represents
+		ET_IMPLEMENTATION_SPECIFIC_METADATA represents a special value used for things like terminating lists of this enum
+
+		Pay attention that an Asset type represents one single bit, so there is a limit to 64 bits.
+
+	*/
+	
     enum E_TYPE : uint64_t
     {
-        //! asset::ICPUBuffer
-        ET_BUFFER = 1u<<0u,
-        //! asset::CImageData - maybe rename to asset::CSubImageData
-        ET_SUB_IMAGE = 1u<<1u,
-        //! asset::ICPUTexture
-        ET_IMAGE = 1u<<2u,
-        //! asset::ICPUMeshBuffer
-        ET_SUB_MESH = 1u<<3u,
-        //! asset::ICPUMesh
-        ET_MESH = 1u<<4u,
-        //! asset::ICPUSkeleton - to be done by splitting CFinalBoneHierarchy
-        ET_SKELETON = 1u<<5u,
-        //! asset::ICPUKeyframeAnimation - from CFinalBoneHierarchy
-        ET_KEYFRAME_ANIMATION = 1u<<6u,
-        //! asset::ICPUShader
-        ET_SHADER = 1u<<7u,
-        //! asset::ICPUSpecializedShader
-        ET_SPECIALIZED_SHADER = 1u<<8,
-        //! asset::ICPURenderpassIndependentPipeline
-        ET_GRAPHICS_PIPELINE = 1u<<9u,
-        //! reserved, to implement later
-        ET_SCENE = 1u<<10u,
-        //! asset::ICPUComputePipeline
-        ET_COMPUTE_PIPELINE = 1u<<11,
-        //! asset::ICPUTextureView
-        ET_TEXTURE_VIEW = 1u<<12,
-        //! asset::ICPUDescriptorSetLayout
-        ET_DESCRIPTOR_SET_LAYOUT = 1u<<13,
-        //! asset::ICPUDescriptorSet
-        ET_DESCRIPTOR_SET = 1u<<14,
-        //! asset::ICPUPipelineLayout
-        ET_PIPELINE_LAYOUT = 1u<<15,
-        //! asset::ICPUBufferView
-        ET_BUFFER_VIEW = 1u<<16,
-        //! asset::ICPUSampler
-        ET_SAMPLER = 1u<<17,
-        //! lights, etc.
-        ET_IMPLEMENTATION_SPECIFIC_METADATA = 1u<<31u
+        ET_BUFFER = 1u<<0u,								//!< asset::ICPUBuffer
+        ET_SUB_IMAGE = 1u<<1u,						    //!< asset::CImageData - maybe rename to asset::CSubImageData
+        ET_IMAGE = 1u<<2u,								//!< asset::ICPUTexture
+        ET_SUB_MESH = 1u<<3u,							//!< asset::ICPUMeshBuffer
+        ET_MESH = 1u<<4u,								//!< asset::ICPUMesh
+        ET_SKELETON = 1u<<5u,							//!< asset::ICPUSkeleton - to be done by splitting CFinalBoneHierarchy
+        ET_KEYFRAME_ANIMATION = 1u<<6u,					//!< asset::ICPUKeyframeAnimation - from CFinalBoneHierarchy
+        ET_SHADER = 1u<<7u,								//!< asset::ICPUShader
+        ET_SPECIALIZED_SHADER = 1u<<8,					//!< asset::ICPUSpecializedShader
+        ET_MESH_DATA_DESCRIPTOR = 1u<<8,				//!< asset::ICPUMeshDataFormatDesc
+        ET_GRAPHICS_PIPELINE = 1u<<9u,					//!< reserved, to implement later
+        ET_SCENE = 1u<<10u,								//!< reserved, to implement later
+        ET_COMPUTE_PIPELINE = 1u<<11,                   //!< asset::ICPUComputePipeline
+        ET_TEXTURE_VIEW = 1u<<12,                       //!< asset::ICPUTextureView
```

Comment:
the order of those is upside down and all over the place

Reply:
yea, in todo for those issues we talked yesterday, will be done today